### DISCUSSION TITLE
validate xml documents error

### DISCUSSION THREAD

**Discussion by trodin** [INSPIRE-MIF/helpdesk-validator#1244] [Ideas]
Hello,

We have validation error in validating dataset for Population distribution:
Conformance class: INSPIRE GML application schemas, General requirments
Schema validation, validate XML documents saying:

XML document '2021_demography - housholds on municipality level_surface.gml': http://inspire.ec.europa.eu/schemas/pd/4.0/PopulationDistributionDemography.xsd, 2:35: s4s-elt-character: Non-whitespace characters are not allowed in schema elements other than 'xs:appinfo' and 'xs:documentation'. Saw '301 Moved Permanently'.

When I open schema: https://[inspire.ec.europa.eu/schemas/pd/4.0/PopulationDistributionDemography.xsd](https://inspire.ec.europa.eu/schemas/pd/4.0/PopulationDistributionDemography.xsd)
in the root of the document we have xmlns:sc="http://www.interactive-instruments.de/ShapeChange/AppInfo"

But this link is not working: https://www.interactive-instruments.de/de/ShapeChange/AppInfo

Is that the cause of the validation error?
![schema validation error](https://github.com/user-attachments/assets/94fa7d10-28a9-45b8-bd39-dd1d8e472058)

Thank you
Tanja

**Comment by fabiovinci** [comment]
Dear @trodin,

this should not be the cause of the validation error since the PopulationDistributionDemography.xsd is valid.

Could you please send us your GML file so that we can take a look?

**Comment by trodin** [comment]
File size too big: 25 MB are allowed, 29 MB were attempted to upload.

can I send it to some mail?

Thanks,
Tanja

**Comment by fabiovinci** [comment]
You could generate a sample dataset, by reducing the number of feature members, or send the whole dataset by email (you can find it on [my user page](https://github.com/fabiovinci)).

**Comment by trodin** [comment]
[2021_demography by gender and agepopulation on county level_TR1novo.zip](https://github.com/user-attachments/files/16347631/2021_demography.by.gender.and.agepopulation.on.county.level_TR1novo.zip)

I found this one smaller.

**Comment by fabiovinci** [comment]
Dear @trodin,

I validated your dataset, selecting both application schemas PD and SU, and I didn't detect the error you reported. See related report: https://inspire.ec.europa.eu/validator/test-run/details.html?id=EIDac19c162-e1a9-470b-91ec-f31447bcfe4c

In addition, test name is now different:

![image](https://github.com/user-attachments/assets/73e06f79-17a1-49e6-998a-a5e13e912324)

Which version of the validator did you use?

**Comment by trodin** [comment]
Ok, thank you.

I use some internal ETF validator on our infrastructure, not the online version since my data is too big for on line version.

Thank you for validating.

Regards,
Tanja

**Comment by trodin** [comment]
Is there any other error, we had one more error related to geometry:
![image](https://github.com/user-attachments/assets/c6ff2430-3906-4f84-96f2-5affd441c15d)
 do you have that?

Thanks,
Tanja

**Comment by fabiovinci** [comment]
Dear @trodin,

I confirm that the error mentioned was also present.

Fina attached the report: [Test run on 13_45 - 23.07.2024 with test suite Annex III - Statistical units (SU) - IOS 1100.html.zip](https://github.com/user-attachments/files/16411462/Test.run.on.13_45.-.23.07.2024.with.test.suite.Annex.III.-.Statistical.units.SU.-.IOS.1100.html.zip)

**Comment by trodin** [comment]
Ok, now I see. But why The feature geometry is not a valid GML geometry? 

And the first error is alos present in your validation report. It is in Conformance class: GML aplication schemas, Statistical Units and PD
it is in test: Validation against INSPIRE official schema, su-gml.b1: validate XML documents against official schema:

How to correct that?
![image](https://github.com/user-attachments/assets/45ac35b8-5d43-40d2-b366-530dfe12ffcd)

**Comment by fabiovinci** [comment]
Regarding the geometry error "_The feature geometry is not a valid GML geometry. Error detected: Invalid polygon. Outer ring of polygon is clockwise within element Surface,_", you should fix it in the source dataset or during the generation of the gml file.

The errors in the test "Validation against INSPIRE official schema" should be due to the presence of two application schemas.

On the contrary, the test "Validation against declared schema" is passed.

![image](https://github.com/user-attachments/assets/721faba1-938f-4279-bebb-ec2c92076413)

We will further investigate if it's possible to avoid errors related to the presence of feature types from different application schemas.

**Comment by fabiovinci** [comment]
Dear all,
since this issue requires the development of new features that are currently unplanned, it will be moved to the related discussion section.
